cds/modules/oauthclient/cern_openid.py

```diff
@@ -178,10 +178,19 @@ def _account_info(remote, resp):
             resp,
         )
 
-    email = resource["email"]
-    external_id = str(resource["cern_uid"])
-    nice = resource["preferred_username"]
-    name = resource["name"]
+    email = resource.get("email")
+    if not email:
+        raise OAuthCERNRejectedAccountError("No email in userinfo", remote, resp)
+
+    external_id = str(resource.get("cern_uid") or resource.get("sub"))
+    if not external_id:
+        raise OAuthCERNRejectedAccountError("No external_id in userinfo", remote, resp)
+
+    raw_username = resource.get("preferred_username") or email
+    if "@" in raw_username:
+        raw_username = raw_username.split("@")[0]
+    nice = raw_username
+    name = resource.get("name") or nice
```

Comment:
I understand that non-cern users are missing the name right? I would also add a comment mention this

Reply:
Ok, if is just a fallback then it's fine for me, thanks!